crates/emitter/src/control_structures.rs

```diff
@@ -206,15 +224,30 @@ impl Jump for BreakEmitter {
 
 impl BreakEmitter {
     pub fn emit(&mut self, emitter: &mut AstEmitter) {
-        // break { control, jumpkind }.emit(self)
-        //      -> register break
-        //      -> jump
         let offset = emitter.emit.bytecode_offset();
         emitter.loop_stack.register_break(offset);
         self.emit_jump(emitter);
     }
 }
 
+pub struct ContinueEmitter {
+    pub jump: JumpKind,
+}
+
+impl Jump for ContinueEmitter {
+    fn jump_kind(&mut self) -> &JumpKind {
+        &self.jump
+    }
+}
+
+impl ContinueEmitter {
+    pub fn emit(&mut self, emitter: &mut AstEmitter) {
+        let offset = emitter.emit.bytecode_offset();
```

Comment:
can you add comment about "prepare for non-local jump" ?
here and also in BreakEmitter.

https://searchfox.org/mozilla-central/rev/a707541ff423ade0d81cef6488e6ecfa09273886/js/src/frontend/BytecodeEmitter.cpp#702-840

so far, the only relevant thing done in `leave_lexical_scope` is emitting `Opcode::DebugLeaveLexicalEnv`, that is mostly optional.
so I think this can be landed without any notable breakage.